watcher/src/bin/main.rs

```diff
@@ -149,14 +152,26 @@ impl WatcherSyncThread {
 
             // For now, ignore origin block, as it does not have a signature.
             let syncing_done = watcher
-                .sync_blocks(1, config.max_block_height, false)
+                .sync_blocks(
+                    1,
+                    config.max_block_height,
+                    Some(max_blocks_per_iteration),
+                    false,
+                )
                 .expect("Could not sync signatures");
-            if syncing_done {
-                log::info!(logger, "sync_signatures indicates we're done");
-                break;
-            }
 
-            sleep(config.poll_interval);
+            // Decide next step before continuing based on sync result
+            match syncing_done {
+                SyncResult::AllBlocksSynced => {
+                    log::info!(logger, "sync_signatures indicates we're done");
```

Comment:
Nit: Wrong name
```suggestion
                    log::info!(logger, "sync_blocks indicates we're done");
```

Reply:
This was the original language, I'm on board with this change, but I'd like to do it in a separate PR as this bug causes FOG to break.